plugins/tableselection/plugin.js

```diff
@@ -208,7 +208,8 @@
 			// The selection is inside one cell, so we should allow native selection,
 			// but only in case if no other cell between mousedown and mouseup
 			// was selected.
-			if ( !fakeSelection.dirty && cells.length === 1 ) {
+			// We don't want to clear selection if widget is event target (#1027).
+			if ( !fakeSelection.dirty && cells.length === 1 && !( CKEDITOR.plugins.widget && CKEDITOR.plugins.widget.isDomWidget( evt.data.getTarget() ) ) ) {
```

Comment:
Maybe we could extract the second part to some function as it's repeated several times in the code?

Reply:
It's repeated both here and in `core/selection.js`.